modules/ts/include/opencv2/ts/ts_ext.hpp

```diff
@@ -62,12 +62,14 @@ extern int testThreads;
             #test_case_name, #test_name, NULL, NULL, \
             ::testing::internal::CodeLocation(__FILE__, __LINE__), \
             (::testing::internal::GetTestTypeId()), \
-            ::testing::Test::SetUpTestCase, \
-            ::testing::Test::TearDownTestCase, \
+            parent_class::SetUpTestCase, \
+            parent_class::TearDownTestCase, \
             new ::testing::internal::TestFactoryImpl<\
                 GTEST_TEST_CLASS_NAME_(test_case_name, test_name)>);\
     void GTEST_TEST_CLASS_NAME_(test_case_name, test_name)::TestBody() BODY_IMPL( #test_case_name "_" #test_name ) \
-    void GTEST_TEST_CLASS_NAME_(test_case_name, test_name)::Body()
+    void GTEST_TEST_CLASS_NAME_(test_case_name, test_name)::bodyMethodName()
+
+#define TEST_(test_case_name, test_name, BODY_IMPL) TEST__(test_case_name, test_name, ::testing::Test, Body, BODY_IMPL)
```

Comment:
Is it possible to omit this intermediate macro?
```.cpp
#define TEST(test_cast_name, test_name) \
  TEST__(test_case_name, test_name, ::testing::Test, Body, CV__TEST_BODY_IMPL)
//...
#define BIGDATA_TEST(test_case_name, test_name) \
  TEST__(BigData_ ## test_case_name, test_name, ::testing::Test, Body, CV__TEST_BIGDATA_BODY_IMPL)
//...
#define PERF_TEST(test_case_name, test_name) \
  TEST__(test_case_name, test_name, ::perf::TestBase, PerfTestBody, CV__PERF_TEST_BODY_IMPL)
```
And then rename `TEST__` to `TEST_`?